### PR TITLE
Test books controller

### PR DESCRIPTION
**Pull request by floresv**
### Summary
Add tests for books_controller, in relation with this, I added a shared_context for authenticated_user with devise token auth using create_new_auth_token method to create a new session.

### Trello Card
https://trello.com/c/ldTBrVFG/25-crear-tests-para-rutas-index-y-show-de-book

### CODE REVIEW

File: spec/controllers/api/v1/books_controller_spec.rb [hunk 1]
```diff
@@ -1,4 +1,63 @@
...
+          books, each_serializer: BookSerializer
+        ).to_json
+
+        expect(JSON.parse(response.body)['page']).to eq(JSON.parse(expected))
+      end
+
+      it 'responds with 200 status' do
+        expect(response).to have_http_status(:ok)
+      end
+    end
```
Comment (leamotta) [inline spec/controllers/api/v1/books_controller_spec.rb:25]: You could test that at least a filter works the way it's supposed to. Eg: Creating a few books with a certain genre and then searching for that genre, making sure you only return the books that match

File: spec/controllers/api/v1/books_controller_spec.rb [hunk 2]
```diff
@@ -1,4 +1,63 @@
...
+    context 'When show a book rented' do
+      let!(:book) { create(:book_rented) }
+
+      before do
+        get :show, params: { id: book.id }
+      end
+
+      it 'responses with a book json' do
+        expected = ShowBookSerializer.new(book).to_json
+        expect(JSON.parse(response.body)).to eq(JSON.parse(expected))
```
Comment (leamotta) [inline spec/controllers/api/v1/books_controller_spec.rb:77]: this does not guarantee that you rendered the rent correctly. Try to go one step deeper inside the response and check if the rent in that body is the same of the book you are creating

File: spec/controllers/api/v1/books_controller_spec.rb [hunk 3]
```diff
@@ -1,4 +1,63 @@
...
+      end
+
+      it 'responds with 200 status' do
+        expect(response).to have_http_status(:ok)
+      end
+    end
```
Comment (leamotta) [inline spec/controllers/api/v1/books_controller_spec.rb:84]: You could also test what happens when you search for a book that doesn't exist.
Always keep in mind the worst case scenario, they are very helpful in the long run when you are refactoring stuff

File: spec/factories/book_factory.rb [hunk 4]
```diff
@@ -7,4 +7,10 @@
     publisher { Faker::Book.publisher }
     year      { Faker::Number.between(1950, Time.zone.today.year) }
   end
+
+  factory :book_rented, parent: :book do
+    after(:create) do |book|
+      create(:rent, book: book)
```
Comment (leamotta) [inline spec/factories/book_factory.rb]: great idea! though it's better to handle this with traits, you can read about that here https://robots.thoughtbot.com/remove-duplication-with-factorygirls-traits

**Review by leamotta — CHANGES_REQUESTED** [review]

File: spec/factories/book_factory.rb (hunk 4 above)
Comment (floresv) [reply spec/factories/book_factory.rb]: I don't have working experience with this, I will read it.

**Review by leamotta — APPROVED** [review]
👍 👍

**Pull request merged**